src/shared/dev-setup.c

```diff
@@ -22,7 +22,7 @@ int lock_dev_console(void) {
         _cleanup_close_ int fd = -EBADF;
         int r;
 
-        fd = open_terminal("/dev/console", O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW);
+        fd = open_terminal("/dev/console", O_RDONLY|O_CLOEXEC|O_NOCTTY);
```

Comment:
Maybe add a comment here:

/* NB: We do not use O_NOFOLLOW here, because some container managers might place a symlink to some pty in /dev/console, in which case it should be fine to lock the target TTY */